tests/downloads/test_api.py

```diff
@@ -134,3 +134,38 @@ async def test_download_cache_reads(error, paired, tmpdir, spawn_client, resp_is
         return
 
     assert resp.status == 200
+
+
+@pytest.mark.parametrize("error", [None, "404"])
+async def test_download_bowtie2_files(error, tmpdir, spawn_client):
+    client = await spawn_client(authorize=True)
+
+    client.app["settings"]["data_path"] = str(tmpdir)
+
+    tmpdir.mkdir("subtractions").mkdir("foo").join("subtraction.1.bt2").write("test")
+
+    subtraction = {
+        "_id": "foo",
+        "name": "Foo",
+        "files": [
+            {
+                "name": "subtraction.1.bt2",
+                "size": 1234567,
+                "type": "bowtie2"
+            }
+        ]
+    }
+
+    if error == "404":
+        subtraction["files"] = []
+
+    await client.db.subtraction.insert_one(subtraction)
+
+    resp = await client.get("/download/subtraction/foo/subtraction.1.bt2")
+
+    if error == "404":
+        assert resp.status == 404
+        return
+
+    assert resp.status == 200
```

Comment:
Please figure out how to test that the correct file is downloaded. Check-in with @jakeale as I have asked him to do this also.